core/src/main/java/jenkins/model/Nodes.java

```diff
@@ -256,6 +259,14 @@ public void run() {
                     if (c != null) {
                         c.recordTermination();
                         c.disconnect(OfflineCause.create(hudson.model.Messages._Hudson_NodeBeingRemoved()));
+                        for(Executor executor: c.getAllExecutors()){
+                            executor.interrupt(Result.ABORTED, new CauseOfInterruption() {
+                                @Override
+                                public String getShortDescription() {
+                                    return "Node was removed by " + User.current().getId();
```

Comment:
May cause NPE

Reply:
My bad, I expected that user is system or some logged user. Thanks!

Reply:
Fixed